cdap-ui/app/features/admin/controllers/namespace/templates-ctrl.js

```diff
@@ -74,52 +118,66 @@ angular.module(PKG.name + '.feature.admin')
 
       fetchApi(params).$promise
         .then(function (res) {
-          var pluginProperties = (res.length? res[0].properties: {});
-
-          plugin._backendProperties = pluginProperties;
+          vm.pluginVersions = res;
 
-          fetchConfig();
+          if (vm.isEdit) {
+            vm.plugin = $filter('filter')(res, {artifact: vm.pluginConfig.artifact})[0];
```

Comment:
It would be nice if we could restrict the use of angular filter-filter to html. We could replace this by array filter method.. As simple as something like,
```res.filter( rObj => angular.equals(rObj.artifact, vm.pluginConfig.artifact));```